misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/BTestRunner.java

```diff
@@ -435,7 +436,6 @@ private void executeBeforeEachFunction(Test test, TestSuite suite, ClassLoader c
 
     private void executeBeforeFunction(Test test, TestSuite suite, ClassLoader classLoader, Scheduler scheduler,
                                        AtomicBoolean shouldSkip, AtomicBoolean shouldSkipTest)  {
-        if (!shouldSkip.get() && !shouldSkipTest.get()) {
```

Comment:
If we are to skip the test, we don't need to run the `before` test for that too right?

Reply:
I have added back the condition incase the beforeEach function fails.